Darwin/StitchCore/StitchCore/Core/Auth/Internal/StitchAuthImpl.swift

```diff
@@ -139,40 +140,105 @@ internal final class StitchAuthImpl: CoreStitchAuth<StitchUserImpl>, StitchAuth
     }
 
     /**
-     * Links the currently authenticated user with a new identity, where the identity is defined by the credential
-     * specified as a parameter. This will only be successful if this `StitchUser` is the currently authenticated
-     * `StitchUser` for the client from which it was created.
+     * Logs out the currently authenticated user, and clears any persisted authentication information.
      *
      * - parameters:
-     *     - withCredential: The `StitchCore.StitchCredential` used to link the user to a new
-     *                       identity. Credentials can be retrieved from an
-     *                       authentication provider client, which is retrieved
-     *                       using the `getProviderClient` method on `StitchAuth`.
-     *     - completionHandler: The completion handler to call when the linking is complete.
-     *                          This handler is executed on a non-main global `DispatchQueue`. If the operation is
-     *                          successful, the result will contain a `StitchUser` object representing the currently
-     *                          logged in user.
+     *     - completionHandler: The completion handler to call when the logout is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
      */
-    internal func link(withCredential credential: StitchCredential,
-                       withUser user: StitchUserImpl,
-                       _ completionHandler: @escaping (StitchResult<StitchUser>) -> Void) {
+    public func logout(_ completionHandler: @escaping (StitchResult<Void>) -> Void) {
         dispatcher.run(withCompletionHandler: completionHandler) {
-            return try self.linkUserWithCredentialInternal(withUser: user, withCredential: credential)
+            try self.logoutInternal(withUserId: nil)
         }
     }
-
+    
     /**
-     * Logs out the currently authenticated user, and clears any persisted authentication information.
+     * Logs out of the user with the given userId. The user must exist in the list of all
+     * users who have logged into this application otherwise this will throw a StitchServiveError.
      *
      * - parameters:
-     *     - completionHandler: The completion handler to call when the logout is complete.
+     *     - userId: A String specifying the desired `userId`
+     *     - completionHandler: The completion handler to call when the switch is complete.
      *                          This handler is executed on a non-main global `DispatchQueue`.
      */
-    public func logout(_ completionHandler: @escaping (StitchResult<Void>) -> Void) {
+    public func logoutUserWithId(userId userId: String,
```

Comment:
[minor][style] These methods are public facing and need to be named with parameters in mind, e.g., `logoutUser(withId userId: String)`